backend/src/routes/classroom.ts

```diff
@@ -90,4 +90,109 @@ export const classroomRoute = new Elysia({ prefix: '/classroom' })
                 thumbnail: t.File(),
             }),
         },
+    )
+    .post('/join-classroom', async ({ body, user, session, set }) => {
+        if (!user || !session) {
+            set.status = 401;
+            return {
+                status: 'error',
+                message: 'Unauthenticated, Please sign in and try again',
+            };
+        }
+
+        const joiningCode = (body as { classroomCode: string }).classroomCode;
+
+        const codeRecord = await sql`
+            SELECT classroom_id, expires_at, section
+            FROM classroom_invite_code
+            WHERE code = ${joiningCode}
+        `;
+
+        if (codeRecord.length == 0) {
+            return {
+                status: 'error',
+                message: 'There is no classroom, please recheck the code.',
+            };
+        }
+
+        const expiresAt = new Date(codeRecord[0].expires_at).getTime();
+        const currentTime = new Date().getTime();
+
+        if (expiresAt < currentTime) {
+            return {
+                status: 'error',
+                message: 'This code is expired, please contact your teacher.',
+            };
+        }
+
+        const studentSection = [codeRecord[0].section];
+
+        const userId = user.id;
+
+        const classroomId = codeRecord[0].classroom_id;
+
+        const isAlreadyJoined = await sql`
+            SELECT classroom_id, user_id
+            FROM study
+            WHERE classroom_id = ${classroomId} AND user_id = ${userId}
+        `;
+
+        if (isAlreadyJoined.length != 0) {
+            return {
+                status: 'error',
+                message: 'You have already joined the class.',
+            };
+        }
+
+        await sql`
+            INSERT INTO study 
+                (user_id, section, classroom_id)
+            VALUES
+                (${userId}, ${studentSection}, ${classroomId})
+        `;
+
+        const [slugRecords] = await sql`
+            SELECT slug
+            FROM classroom
+            WHERE id = ${classroomId}
+        `;
+        const slug = slugRecords.slug;
+        return {
+            status: 'success',
+            message: 'You have joined the classroom.',
+            slug: slug,
+        };
+    })
+    .post(
+        '/create-invite-code',
+        async ({ body }: { body: { slug: string; section: number } }) => {
```

Comment:
If you want to valid body request you should use elysia validator, but for type only its fine.

Here's the document about it, [Elysia Body Validation](https://elysiajs.com/validation/schema-type.html#body)

Reply:
fixed